egs/chime4/asr1_multich/run.sh

```diff
@@ -9,16 +9,22 @@
 # general configuration
 backend=pytorch
 stage=0        # start from 0 if you need to start from data preparation
+stop_stage=1000000
 ngpu=1         # number of gpus ("0" uses cpu, otherwise use gpu)
 debugmode=1
 N=0            # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
 verbose=0      # verbose option
 resume=        # Resume the training from snapshot
 
-# feature configuration
-do_delta=false
+# configuration path
+preprocess_conf=conf/preprocess.json
 
 # network architecture
+use_beamformer=true
+blayers=3
+bunits=300
+bprojs=300
+
```

Comment:
how about adding an option to pass the reference channel. something like negative values for attention based ref or else channel id [0-5] for chime4?

Also I see other beamformer implementations like "mvdr2" and "gev" but maybe they can't be used while training because of the argmax in it but maybe the intention to include it was to use while inference as they have the same learnable parameters. How about an option like "bftype" to use them?

Reply:
> how about adding an option to pass the reference channel. something like negative values for attention based ref or else channel id [0-5] for chime4?
O.K. I'll added.

`mvdr2` and `gev` can't work because there are no  implementation for eigen value decomposition. 
(`argmax` is no problem for training in this case because it's used for getting principal eigen vector and it doesn't require differentiable.)

Actually, I won't to enable these beamformer now because I believe these are no big difference about the performance.

Reply:
Then, we may exclude them for now.

Reply:
I added an option for ref channel and deleted mvdr2 and gev.